docs/low-code-spark/spark-streaming/streaming-source-gem/streaming-event-gem.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Event-based
+id: streaming-event-apps
+description: Event-based Source and Target Gems for Streaming Data Applications
+sidebar_position: 1
+tags:
+  - spark
+  - streaming
+  - kafka
+---
+
+## Event-based Sources and Targets
+
+Prophecy supports **Kafka Streaming** Source and Target. More information on supported Kafka Source and Target options are available [here](https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html).
+
+The Kafka Gem allows inferring the schema of the events by automatically populating the `value` column. Schema inference works with both JSON or AVRO file formats. A user is required to provide an example event for schema inference.
+
+## Create a Kafka Source Gem
+
+A Kafka Source Gem allows the Streaming Pipeline continuously pull data data from a Kafka topic. The users are able to define the following:
+
+- **Broker List** allows users to enter multiple Kafka brokers separated by commas. For example: `kdj-ibg1.us-east-2.aws.cloud:9092, kdj-ibg2.us-east-2.aws.cloud:9092,kdj-ibg3.us-east-2.aws.cloud:9092`
+- **Group ID** is an optional field that allows users to input their consumer group ID. If consumer groups are not utilized, leave this field blank
+- **Session Timeout** is the timeout used to detect client failures when using Kafka's group management facility
+- **Security Protocol** only SASL_SSL is supported at this time
+- **SASL/SCRAM** a family of SASL mechanisms that addresses the security concerns with traditional mechanisms that perform username/password authentication like PALIN. `SCRAM-SHA-256` and `SCRAM-SHA-512` are supported
+
+### Credentials
+
+- **Username and Password (not recommended)**: Kafka cluster user name and password. API key can also be used as Username and API secret as the Password
+- **Databricks Secrets (recommended)**: You may use Databricks to manage your credentials
+- **Kafka Topic**: Name of the topic that the Kafka Source is pulling data from
+
+:::caution
+Typing the username and password into the Prophecy UI is **NOT RECOMMENDED**. This option is provided only for demonstration purposes; this option will commit credentials to the code. It is not a safe option.
+
+The recommended method is to use Databricks Secrets.
+:::
+
```

Comment:
The section below should be the introduction for this page: 
http://localhost:3000/category/streaming-source--target

But I am not sure how to do this.

Reply:
I found an acceptable solution
https://github.com/SimpleDataLabsInc/prophecy-docs/pull/89/commits/605e77abab1e56e17e0d281f25b14c8b177c04c1